src/main/scala/viper/gobra/frontend/info/implementation/typing/StmtTyping.scala

```diff
@@ -120,4 +125,19 @@ trait StmtTyping extends BaseTyping { this: TypeInfoImpl =>
 
     case s => violation(s"$s was not handled")
   }
+
+  /**
+    * Checks whether a struct whose type has name 'name' is cyclically defined (in terms of itself or other cyclic structures).
+    * `visitedTypes` keeps track of the types that were already discovered and checked, used for detecting cyclic definition chains
+    */
+  private def cyclicStructDef(struct: PStructType, visitedTypes: Set[String] = Set.empty) : Boolean = {
```

Comment:
`Set.empty` seems to be a bad default value as the declared name should initially be contained in the set

Reply:
Not necessarily if you are checking a structure which has no assigned name (e.g. a struct literal)

Reply:
Even in such a case, a struct may be invalid if it has a field of a type which is itself an invalid recursive struct

Reply:
Makes sense but shouldn't this check then be performed in `wellDefActualType` with `name` being `None`?

Reply:
Yes, I also realized that and I am changing it :)

Reply:
Just added in latest commit